lib/hanami/cli/commands/app/generate/relation.rb

```diff
@@ -9,18 +9,38 @@ module Generate
           # @api private
           class Relation < Command
             argument :name, required: true, desc: "Relation name"
+            option :gateway, desc: "Generate relation for gateway"
 
             example [
               %(books               (MyApp::Relation::Book)),
               %(books/drafts        (MyApp::Relations::Books::Drafts)),
               %(books --slice=admin (Admin::Relations::Books)),
+              %(books --slice=admin --gateway=extra (Admin::Relations::Books)),
             ]
 
             # @since 2.2.0
             # @api private
             def generator_class
               Generators::App::Relation
             end
+
+            def call(name:, slice: nil, gateway: nil)
+              if slice
+                generator.call(
+                  key: name,
+                  namespace: slice,
+                  base_path: fs.join("slices", inflector.underscore(slice)),
+                  gateway: gateway
+                )
+              else
+                generator.call(
+                  key: name,
+                  namespace: app.namespace,
+                  base_path: "app",
+                  gateway: gateway
+                )
+              end
```

Comment:
Is this whole new stanza necessary just so we can pass through `gateway:`?

(I think I hit this recently too. It's something we should make better, so I just wanted to double check it's for the same reason here too)d

Reply:
@timriley yep i believe this is necessary given the current implementation (probably looks familiar since i copied your implementation lol!) - i would be happy to start thinking of alternative ways to accomplish this